chunky/src/java/se/llbit/chunky/renderer/renderdump/DumpFormat.java

```diff
@@ -0,0 +1,188 @@
+/* Copyright (c) 2021 Chunky contributors
+ *
+ * This file is part of Chunky.
+ *
+ * Chunky is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Chunky is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with Chunky.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package se.llbit.chunky.renderer.renderdump;
+
+import se.llbit.chunky.renderer.scene.Scene;
+import se.llbit.util.TaskTracker;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.function.IntConsumer;
+
+/**
+ * A dump format reads a render dump from a DataInputStream into the scene/
+ * writes a render dump from the scene into a DataOutputStream.
+ *
+ * The input stream is expected to do _not_ contain the magic number and version number -
+ * they should have been read before calling load (logic for that in RenderDump class).
+ * The output stream is expected to contain the magic number and version number -
+ * they should have been written before calling save (logic for that in RenderDump class).
+ *
+ * The "header" of a dump typically contains width, height, spp and renderTime
+ * and is the same for all currently implemented formats.
+ * The strategy for reading/writing the samples (from the buffer in the scene)
+ * has to be implemented.
+ */
+abstract class DumpFormat {
+
+  public void load(
+    DataInputStream inputStream,
+    Scene scene,
+    TaskTracker taskTracker
+  ) throws IOException, IllegalStateException {
+    try (TaskTracker.Task task = taskTracker.task(
+      "Loading render dump",
+      scene.width * scene.height
+    )) {
+      readHeader(inputStream, scene);
+      readSamples(
+        inputStream,
+        scene,
+        pixelProgress -> updateTask(task, scene, pixelProgress)
+      );
+    }
+  }
+
+  protected void readHeader(
+    DataInputStream inputStream,
+    Scene scene
+  ) throws IOException, IllegalStateException {
+    int width = inputStream.readInt();
+    int height = inputStream.readInt();
+    if (width != scene.width || height != scene.height) {
+      throw new IllegalStateException("Scene size does not match dump size");
```

Comment:
`IllegalArgumentException` probably :thinking:

Reply:
It is the internal state of the dump and scene which do not match, the given argument (the scene) should always be correct.
That's my reasoning behind IllegalStateException